### PR TITLE
En 1454 extend listeners

### PR DESCRIPTION
**Pull request by iamishalkin**
listeners have all fields defined in c header files
one more example with all types of subscription added

### CODE REVIEW

File: dxfeed/core/listeners/listener.pyx [hunk 1]
```diff
@@ -131,9 +155,10 @@ cdef void time_and_sale_default_listener(int event_type,
                                                    tns[i].is_valid_tick,
                                                    tns[i].is_eth_trade,
                                                    tns[i].trade_through_exempt,
-                                                   tns[i].is_spread_leg])
+                                                   tns[i].is_spread_leg,
+                                                   tns[i].scope])
 
-CANDLE_COLUMNS = ['Symbol', 'Index', 'Time', 'Sequence', 'Count', 'Open', 'High', 'Low', 'Close', 'Volume', 'VWap',
+CANDLE_COLUMNS = ['Symbol', 'EventFlags', 'Index', 'Time', 'Sequence', 'Count', 'Open', 'High', 'Low', 'Close', 'Volume', 'VWap',
```
Comment (alimantu) [inline dxfeed/core/listeners/listener.pyx:161]: This line is too long even for weak restriction of 120 symbols.

File: dxfeed/core/pxd_include/EventData.pxd [hunk 2]
```diff
@@ -163,7 +163,8 @@ cdef extern from "EventData.h":
         dxf_int_t raw_flags,
         dxf_char_t exchange_code,
         dxf_price_type_t day_close_price_type,
-        dxf_price_type_t prev_day_close_price_type
+        dxf_price_type_t prev_day_close_price_type,
+        dxf_order_scope_t scope
```
Comment (alimantu) [inline dxfeed/core/pxd_include/EventData.pxd:169]: In this file we are using imports using `*` - it's really confusing practice bcs after some time you have absolutely no idea from what particular file which function/Class came from. As mentioned in python Zen 😉 
>Namespaces are one honking great idea -- let's do more of those!

File: pyproject.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = 'dxfeed'
-version = '0.4.0'
+version = '0.4.1'
```
Comment (alimantu) [inline pyproject.toml:3]: We can change it to `0.5.0` here 😉

**Comment by alimantu** [comment]
Here is list to consider:
- [ ]  `TimeNanos` vs `Nanos`
- [ ]  Reorder columns in listeners according to the used structures
- [ ]  Replace `*` in imports with namespace usage

File: dxfeed/core/listeners/listener.pyx [hunk 4]
```diff
@@ -13,7 +13,8 @@ cdef class FuncWrapper:
         return out
 
 
-TRADE_COLUMNS = ['Symbol', 'Price', 'ExchangeCode', 'Size', 'Tick', 'Change', 'DayVolume', 'Time', 'IsETH']
+TRADE_COLUMNS = ['Symbol', 'Sequence', 'Price', 'ExchangeCode', 'Size', 'Tick', 'Change', 'DayVolume',
+                 'DayTurnover', 'Direction', 'Time', 'Nanos', 'RawFlags', 'IsETH', 'Scope']
```
[the remark this answers is not in the document]
Comment (iamishalkin) [reply dxfeed/core/listeners/listener.pyx]: reorder  in later PR

File: dxfeed/core/utils/handler.pyx [hunk 5]
```diff
@@ -102,7 +103,11 @@ cdef class DefaultHandler(EventHandler):
 
 
         df = pd.DataFrame(df_data, columns=self.columns)
-        time_columns = df.columns[df.columns.str.contains('Time')]
+        time_columns = df.columns[df.columns.str.contains('Time')].tolist()
```
Comment (alimantu) [inline dxfeed/core/utils/handler.pyx:106]: Why don't simply use 2 conditions like `a.columns.str.contains('Time') & ~a.columns.str.contains('Nanos')`?
Then we don't need additional variable `nanos` and pop operation.

**Pull request merged**